platform-api/src/routers/auth.py

```
@@ -0,0 +1,44 @@
+from models.settings import WorkerSettings
+from dependencies import get_settings, get_db
+
+router = APIRouter(prefix = "/auth", tags=["Authentication"])
+
+CACHE_EXPIRY_SECONDS = 5 * 60 # 5 minute expiry time
+
+@router.post("/token", tags=["Authentication"])
+async def authenticate_for_access_token(
+    username: str = Form(...),
+    password: str = Form(...),
+    settings: WorkerSettings = Depends(get_settings), 
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)